File: Python3/203. Remove Linked List Elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        
        while head and head.val == val:
            head = head.next
        
        last = ListNode(None)
        last.next = head
        cur = last.next
        
        while cur:
            
            #print(cur.val, last.val)
            if cur.val == val:
                cur = cur.next
                last.next = cur
            else:
                cur = cur.next
                last = last.next
            #print("head = ", head.val)
            
        return head

File: Java/203. Remove Linked List Elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        while(head != null && head.val == val){
            head = head.next;
        }
        
        ListNode last = new ListNode();
        last.next = head;
        ListNode cur = new ListNode();
        cur = last.next;
        
        while(cur != null){
            if(cur.val == val){
                cur = cur.next;
                last.next = cur;
            }
            else{
                cur = cur.next;
                last = last.next;
            }
        }
        
        return head;
    }
}